### DISCUSSION TITLE
Running cucumber in artillery.io

### DISCUSSION THREAD

**Discussion by CHCSkedroski** [artilleryio/artillery#1239] [Artillery]
I was wondering if it is possible to integrate artillery with cucumber so that you can run cucumber based scenarios with artillerys reporting and virtual user concepts

**Comment by hassy** [comment]
I haven't seen anything that integrates Artillery & Cucumber, but it may be possible using Artillery's [engine API](https://www.artillery.io/docs/guides/guides/extension-apis#engines).

Is the idea that you already have a test suite written in Cucumber, and you'd like to run those tests as load tests with Artillery?

Would love to understand the use case better!

**Reply by CHCSkedroski** [reply]
yes that is correct we have some BDD and user workflows written in cucumber and would like to run those with load in artillery

**Reply by hassy** [reply]
got it, thanks! and are those tests using Selenium under the hood? in theory, if the tests only depend on what can be installed from `npm`, it should be possible to run them in Artillery. (for example our new [`playwrigtht` engine](https://github.com/artilleryio/artillery-engine-playwright) takes advantage of that to run load tests with real browsers)

**Reply by CHCSkedroski** [reply]
yes they would be using selenium under the hood thanks for the insight

**Reply by vikramgupta0608** [reply]
any update on this issue , do we able to integrate artillery with cucumber ?

**Reply by bernardobridge** [reply]
Hello @vikramgupta0608 ! We don't currently support `cucumber` out of the box, as Artillery has its own DSL in YAML.

But as Hassy mentioned, it would be possible for you to write a custom engine that takes a Cucumber scenario and runs it, rather than using the HTTP engine by default. The Playwright engine (now lives [here](https://github.com/artilleryio/artillery/tree/main/packages/artillery-engine-playwright)) mentioned above would be a good starting point - it exposes a `flowFunction` API that has access to the Playwright API. You could try something similar using `cucumber-js`.

**Comment by Husain-Syedali** [comment]
Is there any update on the above cucumber query?
 Please let me know if artillery supports cucumber suites, I am looking forward to execute all my playwright+Cucumber based suites using artillery.

**Reply by hassy** [reply]
There's no support for using Cucumber and Playwright with Artillery at the moment, or plans to add it. It may be possible to do via Artillery's extension APIs but it's not something we're looking at doing as a core feature or integration.
